performance_test/thread.py

```diff
@@ -0,0 +1,92 @@
+import time
+import os
+
+from google.cloud import storage
+
+RAW_BUCKET = "sie-raw-images"
+STIMULI_BUCKET = "sie-stimuli-images"
+
+
+class thread:
+    """
+    This is the thread class which uploads image to GCP and record the
+    finishing timee for the pipeline.
+    """
+
+    def __init__(self, participant_id, sleep_time, start_queue, end_queue, threshold):
+        """
+        The function initializes a thread object.
+        Args:
+            participant_id: The id of the participant
+            sleep_time: Sleeping time between request to
+                        check whether image processing is finished.
+            start_queue: A queue to store the starting time for each thread.
+            end_queue: A queue to storee the ending time for each thread.
+            threshold: A threshold to determine whether image processing is finished.
+        Returns:
+            None;
+        """
+        self.participant_id = participant_id
+        self.sleep_time = sleep_time
+        self.start_queue = start_queue
+        self.end_queue = end_queue
+        self.threshold = "0" + str(threshold) if threshold < 100 else str(threshold)
+        self.blob_name = f"rcic_mnes_1_00{self.threshold}_ori.jpg"
+
+    def run(self, file_dir, file_name):
+        """
+        The function runs a thread including steps like upload image to the
+        sie-raw-image bucket and check if certain image is generated in the
+        sie-stimuli-image bucket.
+        Args:
+            file_dir: The local directory of the images location.
+            file_name: The name of the image file.
+        Returns:
+            None;
+        """
+        start_time = time.time()
+        self.start_queue.put(start_time)
```

Comment:
is a queue only used to put and get one start_time and end_time? Since you are using ThreadPoolExecutor, you can return a future then you won't need to store times in a queue

Reply:
Yeah, I can change that.